### PR TITLE
Incremented numeric-string should change to int/float

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/10122
closes https://github.com/phpstan/phpstan/issues/10187

https://3v4l.org/7J17g

### CODE REVIEW

File: tests/PHPStan/Analyser/data/bug-10122.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+namespace Bug10122;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @param string $s
+ * @param numeric-string $ns
+ */
+function doFoo(string $s, $ns, float $f) {
+	assertType('string', ++$s);
```
Comment (zonuexe) [inline tests/PHPStan/Analyser/data/bug-10122.php]: `++$s` will be an int even if the input value is a string. https://3v4l.org/i4rru

refs https://github.com/phpstan/phpstan/issues/10187

Comment (staabm) [reply tests/PHPStan/Analyser/data/bug-10122.php]: great catch, thx

Comment (staabm) [reply tests/PHPStan/Analyser/data/bug-10122.php]: please have another look, I think it should look better now

**Comment by staabm** [comment]
I have tested locally that the error in https://github.com/phpstan/phpstan/issues/10187 no longer happens as requested.
if required I can send a strict-rules `UselessCastRuleTest` regression test, but it feels not necessary.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Please add regression test for 10187.

File: resources/functionMap_php80delta.php [hunk 2]
```diff
@@ -102,7 +102,7 @@
 		'str_ends_with' => ['bool', 'haystack'=>'string', 'needle'=>'string'],
 		'str_starts_with' => ['bool', 'haystack'=>'string', 'needle'=>'string'],
 		'strchr' => ['string|false', 'haystack'=>'string', 'needle'=>'string', 'before_needle='=>'bool'],
-		'stripos' => ['int|false', 'haystack'=>'string', 'needle'=>'string', 'offset='=>'int'],
+		'stripos' => ['0|positive-int|false', 'haystack'=>'string', 'needle'=>'string', 'offset='=>'int'],
```
Comment (staabm) [inline resources/functionMap_php80delta.php:105]: changed to have the same return type across php versions, so we don't need to ignore different errors per php version

**Comment by staabm** [comment]
took me a few attempts to get this work across php versions.. but it should work now

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**